### PR TITLE
Reuse qos port for metrics

### LINKED ISSUE #11649: Observability task: Reuse Meter port to QOS port
- [ ] I have searched the [issues](https://github.com/apache/dubbo/issues) of this repository and believe that this is not a duplicate.

## Describe the proposal

EN:
Observability task: Reuse Meter port to QOS port. With more and more built-in functions of dubbo, more and more ports are exposed. Many ports can be used. At the same time, it can also reduce the creation of HttpServer objects. Ports targeted at the observability index buried point can serve the current QOS service ports, which can be distinguished by path。

中文：
可观测性任务：复用Meter端口到QOS端口，随着dubbo内置功能越来越多对外暴漏的端口也越来越多，很多端口是可以服用，同时也可以减少HttpServer对象的创建，针对可观测性指标埋点的端口可以服用当前QOS服务的端口，通过路径来区分。

### PR DESCRIPTION
**Pull request by wxbty**
## What is the purpose of the change

Issue Number: close #11649 

## Brief changelog
1. The one-time created metricsReporter needs to be registered to beanFactory, so that the BaseCommand spi implementation of qos can obtain.
2. exportHttpServer is no longer required
3. The metrics report is at the application level, so the PrometheusMetricsReporterCmd at the frame level needs to traverse the AppModel, and then obtain the corresponding beanFactory implementations.
4. Http path of qos needs to be modified to support two-level paths, such as xxx:22222/metrics/app1

### CODE REVIEW

File: dubbo-metrics/dubbo-metrics-prometheus/src/main/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricsReporterCmd.java [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+            return getResponseByApplication(applicationModel);
+        } else {
+            String appName = args[0];
+            Optional<ApplicationModel> modelOptional = frameworkModel.getApplicationModels().stream().filter(applicationModel -> appName.equals(applicationModel.getApplicationName())).findFirst();
+            if (modelOptional.isPresent()) {
+                return getResponseByApplication(modelOptional.get());
+            } else {
+                return "Not exist application: " + appName;
+            }
+        }
```
Comment (AlbumenJ) [inline dubbo-metrics/dubbo-metrics-prometheus/src/main/java/org/apache/dubbo/metrics/prometheus/PrometheusMetricsReporterCmd.java:55]: can we return all of the applications

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by songxiaosheng** [comment]
Whether it is possible to optimize the code that the response results are printed to the log. The indicator results will be printed to the log more often,
The DefaultCommandExecutor will print all the indicator results to the log file (where the indicator content is too long, the log result can be filtered or shortened in another way)

是否可以优化下响应结果打印到日志的那段代码，指标结果打印到日志里面会比较多，
DefaultCommandExecutor中会将指标结果全部打印到日志文件中（指标内容太长了log result的地方可以过滤或者换种日志打印方式缩短下）

image

**Comment by wxbty** [comment]
> Whether it is possible to optimize the code that the response results are printed to the log. The indicator results will be printed to the log more often, The DefaultCommandExecutor will print all the indicator results to the log file (where the indicator content is too long, the log result can be filtered or shortened in another way)
> 
> 是否可以优化下响应结果打印到日志的那段代码，指标结果打印到日志里面会比较多， DefaultCommandExecutor中会将指标结果全部打印到日志文件中（指标内容太长了log result的地方可以过滤或者换种日志打印方式缩短下）
> 
> image

Added the option not to print

**Comment by songxiaosheng** [comment]
please add  Metrics request  permission  to  public

**Review by AlbumenJ — APPROVED** [review]

**Pull request merged**